### PR TITLE
skip destructive rebuilds

### PR DESCRIPTION
**Pull request by snopoke**
Opening for review.

There are obviously some risks with this but they aren't destructive risks, mostly that we could get spammed with lots of errors if inserts can't happen (e.g. if a non-nullable column got added).

### CODE REVIEW

File: corehq/ex-submodules/fluff/signals.py [hunk 1]
```diff
@@ -91,7 +91,17 @@ def catch_signal(sender, **kwargs):
     engine.dispose()
 
 
-SimpleDiff = namedtuple('SimpleDiff', 'type table_name item_name')
+@attr.s
+class SimpleDiff(object):
+    type = attr.ib()
```
Comment (mkangia) [inline corehq/ex-submodules/fluff/signals.py]: nit: is it fine to use an inbuilt function name as the name here?

**Review by mkangia — COMMENTED** [review]
Dont have good knowledge of this to review it. Will leave to others pinged on the ticket.

File: corehq/ex-submodules/fluff/signals.py (hunk 1 above)
Comment (snopoke) [reply corehq/ex-submodules/fluff/signals.py]: it's fine as long as it's scoped to an object

File: corehq/apps/userreports/pillow.py [hunk 2]
```diff
@@ -201,15 +202,22 @@ def migrate_tables(self, engine, diffs, table_names, adapters_by_table):
...
+
+        if config.disable_destructive_rebuild and adapter.table_exists:
+            diff_dicts = [diff.to_dict() for diff in diffs]
+            adapter.log_table_rebuild_skipped(source='pillowtop', diffs=diff_dicts)
+            return
+
+        if config.is_static:
             rebuild_indicators.delay(adapter.config.get_id, source='pillowtop')
+        else:
+            adapter.rebuild_table(source='pillowtop')
```
Comment (calellowitz) [inline corehq/apps/userreports/pillow.py]: why this alternate method

File: corehq/apps/userreports/rebuild.py [hunk 3]
```diff
@@ -126,7 +126,7 @@ def add_columns(engine, raw_diffs, table_names):
             col.table = None
             changes[table_name].append({
                 'type': DiffTypes.ADD_COLUMN,
-                'value': col.name
+                'item_name': col.name
```
Comment (calellowitz) [inline corehq/apps/userreports/rebuild.py]: where is this change used?

**Review by calellowitz — APPROVED** [review]
Do not have a ton of context on this code, but overall looks good. Minor questions only

File: corehq/apps/userreports/pillow.py (hunk 2 above)
Comment (emord) [reply corehq/apps/userreports/pillow.py]: It looks like its keeping the previous behavior. I'm guessing that the original intention was that rebuilds (with data) should happen automatically for static data sources. For non-static, report builder or the user should manually trigger rebuilds.

probably deserves some rethinking though not in this PR

**Review by emord — APPROVED** [review]

File: corehq/apps/userreports/pillow.py (hunk 2 above)
Comment (calellowitz) [reply corehq/apps/userreports/pillow.py]: ah yep, i missed the deletion in line 210

File: corehq/apps/userreports/rebuild.py (hunk 3 above)
Comment (snopoke) [reply corehq/apps/userreports/rebuild.py]: This get's stored in the `diff` column of the `DataSourceAction` table so I wanted it to be consistent with `SimpleDiff` which is now also being stored there.

**Pull request merged**